### DISCUSSION TITLE
Proxy Error

### DISCUSSION THREAD

**Discussion by MadZhaan** [Budibase/budibase#5922] [Help]
Hi, 

I wanted to try using the Budibase API to retrieve a row from my table but I keep getting a proxy error. I'm hosting my server locally and the app is already published. Am I using this correctly or is it just not possible for it to retrieve from a local server?

I got this error. "Not found because of proxy error: Error: connect ECONNREFUSED 127.0.0.1:10000"

![Proxy Error](https://user-images.githubusercontent.com/75675418/168752744-aa44174b-85b1-41ad-9d18-d8aeab2655f4.png)

**Comment by melohagan** [comment]
Hi @MadZhaan 

I think you need to use `host.docker.internal` instead of *localhost*

**Reply by melohagan** [reply]
Hey @MadZhaan I initially misunderstood your question - apologies. I thought you were trying to query the API from within the Budibase app. 

If you simply want to grab a row to look at it, then *localhost* will work, but you need to query it from your machine - not the Budibase server.
Any easy way to do that is with Postman: https://www.postman.com/

**Reply by MadZhaan** [reply]
Hey, @melohagan I've tried querying from my machine but now I'm getting a new error. It says Missing/Invalid DB name, status code:500. By the way, I just noticed the reverse proxy section in the documentation. Do I still need to configure a reverse proxy even though I'm using Docker Desktop?

**Reply by melohagan** [reply]
You may be missing the App ID header: `x-budibase-app-id`
<img width="503" alt="Screenshot 2022-05-19 at 08 54 12" src="https://user-images.githubusercontent.com/101575380/169242213-d280cb18-002e-46be-bf47-52d787c14bc0.png">
<img width="391" alt="Screenshot 2022-05-19 at 08 55 48" src="https://user-images.githubusercontent.com/101575380/169242533-13ab4dc5-5aed-457a-82cd-449bb5cfe5b1.png">

**Reply by MadZhaan** [reply]
I'm getting this redirect error again now.
![Screenshot (10)](https://user-images.githubusercontent.com/75675418/169774527-f270a136-6a17-4fbd-a894-46f1b2b68e71.png)
I should be using my internal API key not the API key from the browser page right?

**Reply by aptkingston** [reply]
@MadZhaan you should be using the API key provided by Budibase when clicking on "View API key":
![image](https://user-images.githubusercontent.com/9075550/169793456-44e7ce6b-bda3-4ee5-a5e1-ecb049cc91c8.png)

![image](https://user-images.githubusercontent.com/9075550/169793510-180090ba-1543-4b44-9944-783f76375ce9.png)

Your API key in that screenshot looks too short to me. Where did you get that API key from?

**Reply by MadZhaan** [reply]
@aptkingston I got it from my .env file. It's my internal API key. But I'm still getting the same error even with the given API key.
